lnrpc/rpc.proto

```diff
@@ -1695,6 +1702,20 @@ message Hop {
     can be executed without relying on a copy of the channel graph.
     */
     string pub_key = 8 [json_name = "pub_key"];
+
+    /** 
+    If set to true, then this hop will be encoded using the new variable length
+    TLV format. Note that if any custom tlv_records below are specified, then
+    this field MUST be set to true for then to be encoded properly.
+    */
+    bool tlv_payload = 9;
+
+    /**
+    An optional set of key-value TLV records. This is useful within the context
+    of the SendToRoute call as it allows callers to specify arbitrary K-V pairs
+    to drop off at each hop within the onion.
+    */
+    TLVRecords tlv_records = 10 [json_name = "tlv_records"];
```

Comment:
This shows up in json as
```
 "tlv_records": {
   "records": {
   }
}
```

It would have been nice if there where some kind of type alias for this in protobuf. Or would it be better to not just define `message TLVRecords` and place the map in here directly?

Reply:
I used a new `message` here as it lets us add new fields to `TLVRecords` in the future, vs having in in-line map. We could maybe get some feedback from devs to see if the extra field is annoying or not. FWIW, we have a few other fields that are like this.

Reply:
Do you have a concrete idea of what those new fields to add to `TLVRecords` in the future could be? If not, it would be my preference not to add this extra layer. Suppose it was a list instead of a map, then wouldn't we also just define it in-line?

Reply:
I don't think the layer adds much friction (just another attribute to traverse for devs). If it was a list instead, I would still define it as is now to make things more flexible for the records. 

An example of a fields we may want to add are additional annotations for the fields recognized which are part of the spec, but optional (not routing critical).

Reply:
I don't think qualms about _nesting a proto field_, which at most, costs developers a few characters here and there is worthy of removing the RPC extensions from this PR. Just check out the old keysend PR for the level of demand for the type of flexibility exposed in this PR.

Reply:
I think what is being referred to is that if we wouldn't have custom tlv in the scope of this pr, a migration may not have been necessary and we could think a little longer about the db format. Possibly moving it to full tlv always (including the legacy fields). 

However, there is also the `LegacyPayload` bool that would still have required a migration in the slimmed down pr.